python/src/etos_api/library/validator.py

```diff
@@ -203,15 +268,9 @@ async def validate(self, test_suite_url):
                         test_runners.add(constraint.value)
             docker = Docker()
             for test_runner in test_runners:
-                if test_runner in TESTRUNNER_VALIDATION_CACHE:
-                    timestamp = TESTRUNNER_VALIDATION_CACHE[test_runner]
-                    if (timestamp + TESTRUNNER_VALIDATION_WINDOW) > time.time():
-                        self.logger.info(
-                            "Using cached testrunner validation result: %s", test_runner
-                        )
-                        continue
-                    del TESTRUNNER_VALIDATION_CACHE[test_runner]
-
+                if TestRunnerValidationCache.is_test_runner_valid(test_runner):
```

Comment:
This call needs to be async or it risks hanging everything due to the lock not being asyncio.

Line 45 needs to be an asyncio.Lock instead of threading.Lock.
This call here (line 271) needs to be an await call: await TestRunnerValidationCache.is_test_runner_valid(test_runner) which means that the methods in "TestRunnerValidationCache" must be updated to "async def"

Reply:
Changed to asyncio.